components/ws-manager-mk2/controllers/workspace_controller.go

```diff
@@ -148,6 +148,11 @@ func (r *WorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 func (r *WorkspaceReconciler) actOnStatus(ctx context.Context, workspace *workspacev1.Workspace, workspacePods corev1.PodList) (ctrl.Result, error) {
 	log := log.FromContext(ctx)
 
+	if workspace.Status.Phase != workspacev1.WorkspacePhaseStopped && !r.metrics.containsWorkspace(workspace) {
+		// If the workspace hasn't stopped yet, and we don't know about this workspace yet, remember it.
+		r.metrics.rememberWorkspace(workspace, nil)
```

Comment:
this will construct a `metricState` for this workspace with the assumption that any existing conditions have already been reported on.

E.g. if the workspace already has a BackupFailed condition, no backup failure metric will get reported for it. It assumes it's already been reported before ws-manager-mk2 restarted.

We could potentially loose a few events due to this, but the alternative is re-reporting every condition for all existing workspaces after ws-manager-mk2 restarts